cogs/birthday/birthday.py

```diff
@@ -512,52 +543,100 @@ async def setSelfBirthday(self, ctx: Context, *, birthday: MonthDayConverter):
         birthdayStr = "{0:%B} {0:%d}".format(birthday)
 
         if birthdayConfig:
-            confirmationStr = (
-                f"Are you sure you want to set your birthday to "
-                f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}? "
-                "Only administrators and moderators can reset your birthday afterwards. "
-                f"Type {bold('`yes`', escape_formatting=False)} to confirm."
+            confirmationStr = "\n".join(
+                (
+                    f"Are you sure you want to set your birthday to "
+                    f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}? "
+                    "Only administrators and moderators can reset your birthday afterwards.",
+                    f"Type {bold('`yes`', escape_formatting=False)} to confirm.",
+                )
             )
 
-            await ctx.author.send(f"{headerWarn}: {confirmationStr}")
+            async def mainFlow(destination, carefree: bool = False):
+                # define the time for the message to live for
+                SENSITIVE_MSG_TTL = None if carefree else 10.0
+                try:
+                    await destination.send(
+                        f"{headerWarn}: {confirmationStr}",
+                        delete_after=SENSITIVE_MSG_TTL,
+                    )
+                except discord.Forbidden:
+                    # catch failed DM
+                    raise
+                # wait for answer
+                responseChannel = None
+                if carefree:
+                    responseChannel = destination.dm_channel
+                else:
+                    responseChannel = destination.channel
+
+                def check(msg: discord.Message):
+                    return msg.author == ctx.author and msg.channel == responseChannel
+
+                try:
+                    response = await self.bot.wait_for("message", timeout=10.0, check=check)
+                except asyncio.TimeoutError:
+                    await destination.send(
+                        f"{headerBad}: You took too long. Not setting your birthday."
+                    )
+                    return
 
-            def check(msg: discord.Message):
-                return msg.author == ctx.author and msg.channel == ctx.author.dm_channel
+                if response.content.lower() != "yes":
+                    await destination.send(f"{headerBad}: Declined. Not setting your birthday.")
+                    return
+                # Set birthday and notify user that their birthday has been set
```

Comment:
nit: 1 extra empty line for readability. The comment line here signifies another logic portion.
micronit: it's up to you but other surrounding comments have first letters not capitalized so might as well do the same for this
```suggestion

                # set birthday and notify user that their birthday has been set
```

Reply:
Fixed